src/components/ChatEntry.js

```diff
@@ -1,22 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
 import PropTypes from 'prop-types';
 
 const ChatEntry = (props) => {
+  // const onLike = () => {
+  //   props.updateLike(props.id);
+  // }
+
+  const [isLiked, setIsLiked] = useState(false);
+
+  const likeFill = isLiked ? '❤️' : '🤍';
+
+  const toggleLiked = () => {
+    setIsLiked(!isLiked);
+  };
+
+  const onLike = () => {
+    props.updateLike(props.id);
+    toggleLiked();
+  };
```

Comment:
So there are couple things here that we want to change to fully utilize the method of Lifting Up State! You have `state` (entries) established in your `App.js` file, and we know that when `state` is changed, React will re-render the component `state` is in **as well as all it's children components the new version of state**. We can you that fact here in the `chatEntry`. Keep in mind, your `props.updatelike` function is changing your `entries` state inside of `App.js` therefore it will trigger the re-render of `App.`, `chatLog` and `chatEntry` with the changed `entries`. And since the `liked` property will be changed with can use that to toggle between the two different hearts like so: 

```javascript 
const likeFill = props.liked ? '❤️' : '🤍';  // lines 11 - 22 can be replaced with this

button className="like" onClick={() => props.updateLike(props.id)}> // line 39 can be this 

```